pandas/core/algorithms.py

```diff
@@ -1397,7 +1410,9 @@ def wrapper(arr, indexer, out, fill_value=np.nan):
     return wrapper
 
 
-def _take_2d_multi_object(arr, indexer, out, fill_value, mask_info):
+def _take_2d_multi_object(
+    arr: np.ndarray, indexer: np.ndarray, out: np.ndarray, fill_value, mask_info
+):
```

Comment:
```suggestion
): -> None
```
(and similarly for some others, while we're here?)

Reply:
im ambivalent about the `-> None` pattern

Reply:
why these are ways to annotate
and make it super obvious what is happening 

we should always type void

Reply:
Updated + green